mesonbuild/coredata.py

```diff
@@ -103,8 +103,10 @@ def printable_value(self) -> T.Union[str, int, bool, T.List[T.Union[str, int, bo
     def validate_value(self, value: T.Any) -> _T:
         raise RuntimeError('Derived option class did not override validate_value.')
 
-    def set_value(self, newvalue: T.Any) -> None:
+    def set_value(self, newvalue: T.Any) -> bool:
+        oldvalue = getattr(self, 'value', None)
```

Comment:
This is technically unsafe because the actual `self.value` could be "None", although in this case it seems we escape intact because None isn't a valid value in any current UserOption subclasses.

Reply:
This is not well typed, but it cannot be none, that's not a valid value in the interpreter.

Reply:
More interestingly is whether validate_value will desugar a valid interpreter value down to None.